python/nistoar/midas/dbio/base.py

```diff
@@ -941,21 +943,53 @@ def user_groups(self) -> frozenset:
             self.recache_user_groups()
         return self._whogrps
 
+    def all_groups_for(self, who) -> frozenset:
+        """
+        Return the frozen set of all groups a user or group belongs to.
+        """
```

Comment:
See comment above in `authorized()`; this function is no longer needed, so we can remove it here.